ui/js/directives/treeherder/clonejobs.js

```diff
@@ -219,8 +222,11 @@ treeherder.directive('thCloneJobs', [
             return visible;
         };
 
-        var addJobBtnEls = function(jgObj, jobList) {
+        var renderJobBtnEls = function(jgObj) {
             var lastJobSelected = ThResultSetStore.getSelectedJob($rootScope.repoName);
+            if (!_.isEmpty(lastJobSelected.job)) {
+                lastJobSelected.el = $('.' + getJobMapKey(lastJobSelected.job));
+            }
```

Comment:
Note that calling jQuery functions isn't inherently expensive/problematic, it's calling them 100, 1000, 10000, 100000 times when they become a big issue. :)